lib/ocarina.dart

```diff
@@ -68,7 +68,7 @@ class OcarinaPlayer {
         'seek', {'playerId': _id, 'position': duration.inMilliseconds});
   }
 
-  Future<int> position() async {
+  Future<int?> position() async {
```

Comment:
The channel method will always return a int so the type here should reflect that, and the return value should be unpacked.

Reply:
In the native layer there is a possibility of returning null; for instance under the guard clause here: https://github.com/erickzanardo/ocarina/blob/3b5e668dce959d481ae445492cc9301534080140/ios/Classes/SwiftOcarinaPlugin.swift#L418

I can unpack it in this method and raise an exception (`PlatformException`) if the int result returned from the native layer is null. Would that be the best approach for this case?

Reply:
I agree on raising an exception. It should never happen for an id be passed to the native layer, and the player don't exist there.